### PR TITLE
feat: support acquiring token by device code

### PR DESCRIPTION
**Pull request by dominique-pfister**
BREAKING CHANGE: refreshToken no longer supported

### CODE REVIEW

**Review by tripodsan — APPROVED** [review]

File: src/OneDriveAuth.js [hunk 1]
```diff
@@ -60,14 +60,17 @@ export class OneDriveAuth {
     if (opts.username || opts.password) {
       throw new Error('Username/password authentication no longer supported.');
     }
+    if (opts.refreshToken) {
+      throw new Error('Refresh token no longer supported.');
+    }
```
Comment (dominique-pfister) [inline src/OneDriveAuth.js:65]: remove refresh token support

File: src/OneDriveAuth.js [hunk 2]
```diff
@@ -217,30 +221,6 @@ export class OneDriveAuth {
...
 
-  /**
-   * Performs a login using an interactive flow which prompts the user to open a browser window and
-   * enter the authorization code.
-   * @params {function} [onCode] - optional function that gets invoked after code was retrieved.
-   * @returns {Promise<AuthenticationResult>}
-   */
-  async acquireTokenByDeviceCode(onCode) {
-    const { log, app } = this;
-    try {
```
Comment (dominique-pfister) [inline src/OneDriveAuth.js:228]: see below

File: src/OneDriveAuth.js [hunk 3]
```diff
@@ -295,10 +275,13 @@ export class OneDriveAuth {
     }
 
     try {
-      if (this.refreshToken) {
-        log.debug('acquire token with refresh token.');
-        return await app.acquireTokenByRefreshToken({
-          refreshToken: this.refreshToken,
```
Comment (dominique-pfister) [inline src/OneDriveAuth.js:301]: removed support for refreshToken

File: src/OneDriveAuth.js [hunk 4]
```diff
@@ -295,10 +275,13 @@ export class OneDriveAuth {
...
+      if (this.onCode) {
+        log.debug('acquire token with device.');
+        return await app.acquireTokenByDeviceCode({
+          deviceCodeCallback: async (code) => {
+            await this.onCode(code);
+          },
+          scopes: this.scopes,
         });
```
Comment (dominique-pfister) [inline src/OneDriveAuth.js:285]: new `acquireTokenByDeviceCode`

**Review by tripodsan — APPROVED** [review]

**Review by tripodsan — APPROVED** [review]

File: test/onedrive-auth.test.js [hunk 5]
```diff
@@ -103,12 +128,72 @@ describe('OneDriveAuth Tests', () => {
...
+        refresh_token: 'dummy',
+        access_token: 'dummy',
+        expires_in: 81000,
+        id_token: new UnsecuredJWT({
+          sub: 'test',
+        }).encode(),
+        client_info: Buffer.from(JSON.stringify({
+          uid: 'Bob',
+          utid: 'common',
+        })).toString('base64'),
```
Comment (dominique-pfister) [inline test/onedrive-auth.test.js:178]: these 2 are required to have the `isAuthenticated()` below return true

**Pull request merged**